lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -29,12 +39,67 @@ class Chef::Provider::AwsEbsVolume < Chef::Provider::AwsProvider
     if existing_volume
       converge_by "Deleting EBS volume #{fqn} in #{new_resource.region_name}" do
         existing_volume.delete
+
+        Retryable.retryable(:tries => 30, :sleep => 2, :on => TimeoutError) do
+          raise TimeoutError,
+            "Timed out waiting for volume #{new_resource.volume_id} to be deleted." if existing_volume.exists?
+        end
       end
     end
 
     new_resource.delete
   end
 
+  action :attach do
```

Comment:
Typically, the way we handle this is more descriptive: we'd handle the attach as part of the `create`:

```ruby
aws_ebs_volume 'volume' do
  ...
  attached_to_instance 'i-234234234'
end
```

And the create action would both create and attach the instance.  No need for a separate `attach` action.

Also, we should let you attach to a `machine` instead of an `instance`.